docs_guidelines/apidocs/api.guidelines.rst

```diff
@@ -0,0 +1,57 @@
+########################
+API Reference Guidelines
+########################
+
+.. note:: 
+   
+   We are mainly using ``jupyter-execute`` directive for putting code examples. We will investigate 
+   using ``doctest`` to replace ``jupyter-execute`` in the future wherever it's possible to reduce 
+   doc building process complexity. See 
+   `qiskit-terra#7661 <https://github.com/Qiskit/qiskit-terra/issues/7661>`_ for discussions.
+
+1. Try to show as many arguments/options as possible within reason
+==================================================================
+
+Focus on the arguments that make bigger differences. Among the options of a same argument, show only
```

Comment:
```suggestion
Focus on the arguments that make bigger differences. 

Among the options of a same argument, show only
```

I'd separate the point about different arguments from the one about different options for a single argument.

Reply:
The problem here is that there are actually two different points being made but both ended up mixed in the same line.

Let `my_func(x, y, z)` be some Qiskit function. The first point was about choosing if it's useful to show the arguments `x`, `y` and `z` or only some of them.

Now imagine that the argument `x` can be set to `"option_1"`, `"option_2"` and `"option_3"`. Then the second point is about which of those options to show, as maybe the difference between `x="option_1"` and `x="option_2"` is not particularly interesting.

I'm probably misunderstanding something or mixing up terminology 😅